lib/ASRouter.jsm

```diff
@@ -674,6 +680,61 @@ class _ASRouter {
     });
   }
 
+  hasGroupsEnabled(groups = []) {
+    return this.state.groups
+      .filter(({ id }) => groups.includes(id))
+      .every(({ enabled }) => enabled);
+  }
+
+  /**
+   * Fetch all message groups and update Router.state.groups
+   * @param provider RS messages provider for message groups
+   */
+  async loadAllMessageGroups(provider) {
+    if (!provider) {
+      return;
+    }
+    let { messages } = await MessageLoaderUtils.loadMessagesForProvider(
```

Comment:
It's a bit confusing that you're loading groups via `loadMessagesForProvider` I would probably have a generic function to like load data i.e. either messages or groups from the provider, and then have anything specific to messages be handled separately